components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/ToolbarPresenter.kt

```diff
@@ -69,10 +70,14 @@ class ToolbarPresenter(
         toolbar.setSearchTerms(session?.searchTerms ?: "")
         toolbar.displayProgress(session?.progress ?: 0)
         updateToolbarSecurity(session?.securityInfo ?: Session.SecurityInfo())
+        session?.let {
+            updateToolbarTrackingProtection(it)
+        }
     }
 
     override fun onUrlChanged(session: Session, url: String) {
         renderer.post(url)
+        updateToolbarTrackingProtection(session)
```

Comment:
Why do we need that here and why isn't that covered by `onTrackerBlockingEnabledChanged` and `onTrackerBlocked`?

Reply:
I was seeing a issue where I switched from a site that has trackers to a site that doesn't have trackers and wasn't getting notified. I will do some further testing to see if there is any relation

Reply:
If some reseting is needed when navigating then this probably needs to happen in `EngineObserver` so that everyone benefits from it and doesn't need to listen to `onUrlChanged()`.

Reply:
👍

Reply:
Now, I realized exactly why do we need this here.

`onTrackerBlockingEnabledChanged` is called when we change tracking protection to **ON** or **OFF**.

and
 
`onTrackerBlocked` is called when a new tracker is blocked.

But when I have a session that has **tracking protection on** and has **some trackers blocked** then I navigate to a new page that doesn't change `trackerBlockingEnabled` **nor has trackers on it.** I don't have a way to know anything about this page. Because we are only observing when we change the tracking protection to **ON** or **OFF** or when new trackers are blocked and none of these cases happen in that scenario.